bundles/http/Citrix/AdobeSign/31fff7a5-ffa8-4019-9e2f-e477837f92de.js

```diff
@@ -0,0 +1,596 @@
+                }
+            ],
+            role: actionParameters.role3,
+            order: actionParameters.order == 'inorder' ? actionParameters.order3 : 1
+        })
+    }
+    const response = await client.fetch('/api/rest/v6/agreements', {
+        method: "POST",
+        headers: {
+            "x-api-user": `userid:${actionParameters.userid}`
+        },
+        body: JSON.stringify(
+            {
+                fileInfos: [
+                    fileInfos
+                ],
+                name: `${actionParameters.agreementName}`,
+                participantSetsInfo: participantSetsInfo,
+                signatureType: 'ESIGN',
+                state: 'IN_PROCESS'
+
+            }
+        )
+    })
+    const json = await response.json()
+    if (!response.ok) {
+        throw new Error(JSON.stringify(json))
+    }
+    const {id} = json
+    let agreement = new Map([[id, { "userid": actionParameters.userid, "name": actionParameters.agreementName }]])
+    await Promise.all([
+        syncAgreements(dataStore, client, [actionParameters.userid], integrationParameters),
+        syncArgreementDetails(client, dataStore, agreement),
+    ])
+}
+
+async function shareAgreement({ dataStore, client, actionParameters}) {
+    const response = await client.fetch(`/api/rest/v6/agreements/${actionParameters.agreementId}/members/share`, {
+        method: 'POST',
+        headers: {
+            "x-api-user": `userid:${actionParameters.userid}`,
+        },
+        body: JSON.stringify({
+            shareCreationInfo: [
+                {
+                    email: actionParameters.email != "" ? actionParameters.email : actionParameters.emailUnknown,
+                    message: actionParameters.message
+                }
+            ]
+        })
+    })
+    if (!response.ok) {
+        throw new Error(await response.text())
+    }
+}
+
+async function cancelAgreement({ dataStore, client, actionParameters, integrationParameters }) {
+    const response = await client.fetch(`/api/rest/v6/agreements/${actionParameters.agreementId}/state`, {
+        method: 'PUT',
+        headers: {
+            "x-api-user": `userid:${actionParameters.userid}`,
+        },
+        body: JSON.stringify({
+            state: "CANCELLED"
+        })
+    })
+
+    if (!response.ok) {
+        throw new Error(await response.text())
+    }
+    let agreement = new Map([[actionParameters.agreementId, { "userid": actionParameters.userid, "name": actionParameters.agreementName }]])
+    await Promise.all([
+        syncAgreements(dataStore, client, [actionParameters.userid], integrationParameters),
+        syncArgreementDetails(client, dataStore, agreement),
+    ])
+}
+async function uploadFiles(dataStore, client, actionParameters) {
+    console.log(JSON.stringify(actionParameters.attachments))
+    const formData = new FormData()
+    if(actionParameters.attachments.length == 0){
```

Comment:
Is it initialize to `[]`?
When I look to line 379, it isn't, probably ->

```suggestion
    if (!actionParameters.attachments || actionParameters.attachments.length === 0){
```

Reply:
changes implemented